autopilot-daemon/network/network_workload.py

```diff
@@ -0,0 +1,149 @@
+from iperf3_utils import *
+
+
+#
+# TODO: Make this an abstract class...
+#
+#
+
+
+class NetworkWorkload:
+    def __init__(self, namespace=None, workload_name="Ring Topology"):
+        self.namespace = namespace or os.getenv("NAMESPACE")
+        self.workload = workload_name
+        self.log = logging.getLogger(__name__)
+        logging.basicConfig(
+            format="[NETWORK] - [WORKLOAD-GEN] - [%(levelname)s] : %(message)s",
+            level=logging.INFO,
+        )
+
+        try:
+            config.load_incluster_config()
+            self.v1 = client.CoreV1Api()
+        except:
+            self.log.error("Failed to load Kubernetes CoreV1API.")
+            exit(1)
+
+    def get_all_ifaces(self):
+        address_map = {}
+
+        try:
+            autopilot_pods = self.v1.list_namespaced_pod(
+                namespace=self.namespace, label_selector="app=autopilot"
+            )
+        except ApiException as e:
+            self.log.error(
+                "Exception when calling CoreV1Api->list_namespaced_pod: %s\n" % e
+            )
+            exit(1)
+
+        for pod in autopilot_pods.items:
+            entrylist = json.loads(
+                pod.metadata.annotations["k8s.v1.cni.cncf.io/network-status"]
+            )
+            if len(entrylist) > 0:
+                for entry in entrylist:
```

Comment:
This function name is misleading. I'm getting all net1-X interfaces. I'm currently ignoring eth0 and eth1 etc.

Reply:
For instance, here's the output for:

autopilot_node_map_json = self.gen_autopilot_node_map_json()

```
{
    "WORK-NAME1": {
        "pod": "autopilot-pod1,
        "endpoint": "Y.Y.Y.Y",
        "netifaces": [
            "X.X.X.X",
             ...15 more rows later,
             "X.X.X.X",
        ]
    },
    "WORK-NAME2": {
        "pod": "autopilot-pod2,
        "endpoint": "Y.Y.Y.Y",
        "netifaces": [
            "X.X.X.X",
             ...15 more rows later,
             "X.X.X.X",
        ]
    },
    "WORK-NAME3": {
        "pod": "autopilot-pod3",
        "endpoint": "Y.Y.Y.Y",
        "netifaces": [
            "X.X.X.X",
             ...15 more rows later,
             "X.X.X.X",
        ]
    }
}
```

Reply:
this will have to go through an update as soon as we make this annotation `pod.metadata.annotations["k8s.v1.cni.cncf.io/network-status"]` configurable through Helm. It can be this way for now